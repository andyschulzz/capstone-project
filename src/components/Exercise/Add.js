import React, { useState } from 'react'
import { useForm } from 'react-hook-form'
import Form from './Form'
import ButtonBackToHome from '../common/ButtonBackToHome'
import ButtonEdit from './ButtonEdit'
import { GreenButton } from '../common/Button'
import * as S from './Form.styles'
import PropTypes from 'prop-types'

Add.propTypes = {
  handleExerciseAdd: PropTypes.func.isRequired,
}

export default function Add({ handleExerciseAdd }) {
  const [disabled, setDisabled] = useState(false)
  const { register, handleSubmit } = useForm()

  return (
    <S.Form data-testid="add-form" id="add" onSubmit={handleSubmit(handleAdd)}>
      <S.ButtonWrapper>
        <ButtonBackToHome />
        {(!disabled && (
          <GreenButton mla inactive>
            Save
          </GreenButton>
        )) || <ButtonEdit />}
      </S.ButtonWrapper>
      <S.Wrapper>
        <Form register={register} disabled={disabled} />
      </S.Wrapper>
    </S.Form>
  )
  function handleAdd(data) {
    Object.keys(data).map(key => (data[key] = data[key].trim()))
    setDisabled(!disabled)
    if (!disabled) {
      handleExerciseAdd(data)
    }
  }
}
