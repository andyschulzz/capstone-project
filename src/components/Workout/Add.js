import React from 'react'
import ExerciseList from './ExerciseList'
import ButtonSave from './ButtonSave'
import { useHistory } from 'react-router-dom'
import * as S from './Add.styles'
import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form'

Add.propTypes = {
  exercises: PropTypes.array.isRequired,
  handleWorkoutAdd: PropTypes.func.isRequired,
  selectedWorkouts: PropTypes.array.isRequired,
}

export default function Add({
  exercises,
  handleWorkoutAdd,
  handleWorkoutTitle,
  selectedWorkouts,
}) {
  const history = useHistory()
  const { register, handleSubmit, formState } = useForm({
    mode: 'onChange',
  })

  return (
    <>
      <S.ButtonWrapper>
        <h3>Build your Routine</h3>
        {selectedWorkouts.length === 0 ? (
          <ButtonSave form={'add'} formState={false} />
        ) : (
          <ButtonSave form={'add'} formState={formState} />
        )}
      </S.ButtonWrapper>
      <form id="add" onSubmit={handleSubmit(onSubmit)}>
        <S.Textarea
          ref={register({ required: true })}
          autoFocus
          type="text"
          name="name"
          id="name"
          placeholder="Name your workout"
        />
      </form>
      <S.Wrapper>
        <ExerciseList
          exercises={exercises}
          handleWorkoutAdd={handleWorkoutAdd}
        />
      </S.Wrapper>
    </>
  )

  function onSubmit(input) {
    handleWorkoutTitle(input.name)
    history.push('/workouts/details')
  }
}
