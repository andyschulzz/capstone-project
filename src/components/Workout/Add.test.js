import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import Add from './Add.js'
import { MemoryRouter as Router } from 'react-router-dom'

const mockHistoryPush = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

const handleWorkoutAddMock = jest.fn()
const handleWorkoutTitleMock = jest.fn()
const exercisesMock = [
  {
    id: '1',
    name: 'Test',
    type: 'Test',
    instructions: 'Test',
    title: 'Test',
  },
]

afterEach(cleanup)

describe('<Add />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <Add
          exercises={exercisesMock}
          handleWorkoutAdd={handleWorkoutAddMock}
          handleWorkoutTitle={handleWorkoutTitleMock}
          selectedWorkouts={[]}
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should show exercises if button is clicked', () => {
    const { getByTestId } = render(
      <Router>
        <Add
          exercises={exercisesMock}
          handleWorkoutAdd={handleWorkoutAddMock}
          handleWorkoutTitle={handleWorkoutTitleMock}
          selectedWorkouts={[]}
        />
      </Router>
    )
    fireEvent.click(getByTestId('add-button'))
    const inputNode = getByTestId('exercise')
    expect(inputNode).toBeVisible()
  })
})
