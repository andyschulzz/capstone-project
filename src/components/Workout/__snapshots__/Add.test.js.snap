// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> should render correctly 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: none;
  border: none;
  outline: none;
  border-radius: 5px;
  margin-left: auto;
  padding: 5px 15px;
  font-size: 1rem;
  height: 36px;
  cursor: pointer;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(200,100%,50%,0.2);
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
}

.c3 span:first-child {
  margin-top: 5px;
}

.c0 {
  overflow-y: auto;
  padding: 0 40px 30px;
}

.c4 {
  margin-top: 20px;
  width: 100%;
  border: none;
  outline: none;
  height: 40px;
  resize: none;
  font-size: 1.4rem;
  background-color: #fff;
  color: #333;
}

.c4::-webkit-input-placeholder {
  font-weight: 300;
}

.c4::-moz-placeholder {
  font-weight: 300;
}

.c4:-ms-input-placeholder {
  font-weight: 300;
}

.c4::placeholder {
  font-weight: 300;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: #fff;
  padding: 40px 0 30px;
}

.c1 h3 {
  margin: 0;
  font-weight: bold;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: none;
  border: none;
  outline: none;
  border-radius: 5px;
  padding: 5px 15px;
  font-size: 1rem;
  height: 36px;
  cursor: pointer;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsl(200,100%,50%);
  margin: 30px;
  text-transform: uppercase;
}

.c5 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

<form
    class="c0"
    data-testid="add-form"
    id="add"
  >
    <div
      class="c1"
    >
      <h3>
        Build Your Routine
      </h3>
      <button
        class="Button-sc-18sujpt-0 c2"
      >
        Next
      </button>
    </div>
    <section
      class="c3"
    >
      <textarea
        class="c4"
        id="name"
        maxlength="30"
        minlength="1"
        name="name"
        placeholder="Name your workout"
        required=""
        type="text"
      />
      <button
        class="Button-sc-18sujpt-0 c5"
        data-testid="add-button"
      >
        Add Exercises
      </button>
    </section>
  </form>
</DocumentFragment>
`;
